src/compiler/utils/Stack.js

```diff
@@ -0,0 +1,33 @@
+let Stack = function() {
+  let history = [],
+      memory = [];
+  let count = 0;
+  function current() {
+    return history.length ? history[history.length - 1] : null;
+  }
+  this.current = current;
+  this.stack = function() {
```

Comment:
Those work. I think `peek` usually refers to seeing the item on top of the stack without `pop`ping it, right?